### PR TITLE
Add rootKeyPath in GarageRequestParameterContainer

### PR DESCRIPTION
**Pull request by hatuyuki4**
Add rootKeyPath in GarageRequestParameterContainer.
It is used for decoding by Himotoki.

### CODE REVIEW

File: Sources/GarageRequestType.swift [hunk 1]
```diff
@@ -4,6 +4,7 @@ import APIKit
 public protocol GarageRequestParameterContainer {
     var method: HTTPMethod { get }
     var path: String { get }
+    var rootKeyPath: [String]? { get }
```
Comment (slightair) [inline Sources/GarageRequestType.swift]: The `rootKayPath` is option parameter for decode process.
So I propose followings.

- introduce `GarageDecodeResponseParameterContainer` and add default implementation.
- change name `rootKeyPath` => `decodeRootKeyPath`
- change variable type of `rootKeyPath` to `KeyPath` (Himotoki)

File: Sources/RequestBuilder.swift [hunk 2]
```diff
@@ -113,12 +119,19 @@ struct MultipleResourceRequest<R: GarageRequest, D: Decodable>: ResourceRequest
...
 
     func response(from object: Any, urlResponse: HTTPURLResponse) throws -> Response {
-        guard let resource: [D] = try? decodeArray(object) else {
-            throw ResponseError.unexpectedObject(object)
-        }
 
         let parameters = headerParameters(from: urlResponse)
-        return GarageResponse(resource: resource, totalCount: parameters.totalCount, linkHeader: parameters.linkHeader)
+        if let rootKeyPath = baseRequest.rootKeyPath {
+            guard let resource: [D] = try! decodeArray(object, rootKeyPath: KeyPath(rootKeyPath)) as [D]? else {
```
Comment (slightair) [inline Sources/RequestBuilder.swift]: It is not necessary to use `try!`.
Maybe, I had been used `try!` in old `swift3` branch and already fixed.

**Review by slightair — CHANGES_REQUESTED** [review]
Thanks! 🍶

File: Sources/GarageRequestType.swift (hunk 1 above)
Comment (hatuyuki4) [reply Sources/GarageRequestType.swift]: Rename to `decodeRootKeyPath` and change variable type.

https://github.com/slightair/GarageClientSwift/pull/3/files#diff-29b941133987425a8183ef2f69e3273cR8

File: Sources/RequestBuilder.swift (hunk 2 above)
Comment (hatuyuki4) [reply Sources/RequestBuilder.swift]: Fix it.
https://github.com/slightair/GarageClientSwift/pull/3/files#diff-1d696d720be823d13aae2986bcc884f5R125

**Comment by hatuyuki4** [comment]
@slightair 
Thank you for reviewing. Fixed any points.

**Comment by slightair** [comment]
OK, Thanks!

**Pull request merged**